### PR TITLE
Addressing #420 (for CCI Toolbox)

### PR DESCRIPTION
**Pull request by forman**
* Slightly changed signature of `xcube.core.store.DataStore.get_dataset_ids()`
  by adding a new keyword argument `include_attrs: Sequence[str] = None` that
  can be used to obtain a minimum set of dataset attributes for each returned 
  dataset identifier. However, `include_attrs` is ignored to far in the "s3", 
  "memory", and "directory" data stores. (#420) 

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `docs/CHANGES.md`
* [ ] AppVeyor and Travis CI passes
* [x] Test coverage remains or increases (target 100%)
* [ ] Associated issues closed after merge

### CODE REVIEW

File: test/core/store/stores/test_directory.py [hunk 1]
```diff
@@ -162,23 +162,23 @@ def test_get_data_ids(self):
...
             {
-                ('cube-1-250-250.zarr', None),
-                ('cube-5-100-200.zarr', None),
-                ('cube.nc', None),
+                ('cube-1-250-250.zarr', {}),
+                ('cube-5-100-200.zarr', {}),
+                ('cube.nc', {}),
             },
-            set(self.store.get_data_ids(include_titles=False))
+            set(self.store.get_data_ids(include_attrs=["title"]))
```
Comment (TonioF) [inline test/core/store/stores/test_directory.py:169]: As this is not the expected behaviour, I recommend to comment this part of the test out as long as the stores do not return titles (or any other attributes).

File: xcube/core/store/store.py [hunk 2]
```diff
@@ -153,22 +153,41 @@ def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
...
-
-        The returned iterator items are 2-tuples of the form (*data_id*, *title*), where *data_id*
-        is the actual data identifier and *title* is an optional, human-readable title for the data.
-        If *include_titles* is false, the second item of the result tuple will be None.
-
-        :param type_specifier: If given, only data identifiers that are available as this type are returned. If this is
-        omitted, all available data identifiers are returned.
-        :param include_titles: If true, the store will attempt to also provide a title.
+        If a store implementation supports only a single data type, it should verify that
+        *type_specifier* is either None or compatible with the supported data type.
```
Comment (TonioF) [inline xcube/core/store/store.py:165]: If a store implementation supports only a single data type, it should verify that *type_specifier* is either None or compatible with the supported data type. -> A store implementation should verify that *type_specifier* is either None or compatible with a supported data type.

File: xcube/core/store/store.py [hunk 3]
```diff
@@ -153,22 +153,41 @@ def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
...
+
+        If *include_attrs* is provided, it must be a sequence of names of metadata attributes.
+        The store will then return extra metadata for each returned data resource
+        identifier according to the names of the metadata attributes as tuples (*data_id*, *attrs*).
+
+        Hence, the type of the returned iterator items depend on the value of *include_attrs*:
```
Comment (TonioF) [inline xcube/core/store/store.py:171]:
```suggestion
        Hence, the type of the returned iterator items depends on the value of *include_attrs*:
```

File: xcube/core/store/store.py [hunk 4]
```diff
@@ -153,22 +153,41 @@ def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
...
+
+        - If *include_attrs* is None (the default), the method returns an iterator
+          of dataset identifiers *data_id* of type `str`.
+        - If *include_attrs* is a sequence of attribute names, even an empty one,
+          the method returns an iterator of tuples (*data_id*, *attrs*) of type
+          `Tuple[str, Dict]`, where *attrs* is a dictionary filled according to the
+          names in *include_attrs*.
+
+        The individual attributes must not exists in the dataset's metadata, they may also be
```
Comment (TonioF) [inline xcube/core/store/store.py:180]: Well, the title is often part of the metadata, I don't see why the two shall be mutually exclusive. I suggest to remove this sentence.

File: xcube/core/store/store.py [hunk 5]
```diff
@@ -153,22 +153,41 @@ def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
...
+        generated on-the-fly. An example for a generic attribute name is "title".
+        A store should try to resolve ```include_attrs=["title"]``` by returning items such as
+        ```("ESACCI-L4_GHRSST-SSTdepth-OSTIA-GLOB_CDR2.1-v02.0-fv01.0.zarr",
+        {"title": "Level-4 GHRSST Analysed Sea Surface Temperature"})```.
+
+        :param type_specifier: If given, only data identifiers that are available as this type are returned.
+            If this is omitted, all available data identifiers are returned.
+        :param include_attrs: A sequence of names of attributes to be returned for each dataset identifier.
+            If given, the store will attempt to also provide the given set of dataset attributes.
```
Comment (TonioF) [inline xcube/core/store/store.py:189]:
```suggestion
            If given, the store will attempt to provide the set of requested dataset attributes in addition to the data ids.
```

**Review by TonioF — CHANGES_REQUESTED** [review]
I like it, but please consider my comments.

**Comment by TonioF** [comment]
One more thing: Four tests fail.

**Comment by TonioF** [comment]
And, I'm sorry: Where do we put the information what attributes a store offers?

File: test/core/store/stores/test_directory.py (hunk 1 above)
Comment (forman) [reply test/core/store/stores/test_directory.py]: This _is definitely_ the expected behaviour for _this_ implementation. I agree, adding a comment would be good. Once we intent to change behaviour, we do this by a test-driven aypproach and _change the test accordingly first_.

**Comment by forman** [comment]
> And, I'm sorry: Where do we put the information what attributes a store offers?

We'll define a set of common and generic attributes that may be requested here. We put them in the method's docstring later.

**Pull request merged**